django/core/urlresolvers.py

```diff
@@ -647,3 +648,34 @@ def is_valid_path(path, urlconf=None):
         return True
     except Resolver404:
         return False
+
+
+def translate_url(url, lang_code, deactivate=False):
+    """
+    Given an URL (absolute or relative), try to get its translated version in
+    the `lang_code` language (either by i18n_patterns or by translated regex).
+    Return the original URL if no translated version is found.
+
+    This function has the side-effect of setting the current language to
```

Comment:
Is it expensive to restore the old language? I'd think it'd be less error prone if the method didn't have side-effects by default and simpler if we could drop the `deactivate` parameter, but I'll let you decide as the i18n expert.